src/group_theory/monoid_localization.lean

```diff
@@ -46,21 +46,27 @@ a structure up to isomorphism, and reason about things that satisfy the predicat
 The infimum form of the localization congruence relation is chosen as 'canonical' here, since it
 shortens some proofs.
 
+To apply a localization map `f` as a function, we use `f.to_map`, as coercions don't work well for
+this structure.
+
 ## Tags
 localization, monoid localization, quotient monoid, congruence relation, characteristic predicate,
 commutative monoid
 -/
-
+set_option old_structure_cmd true
 namespace add_submonoid
 variables {M : Type*} [add_comm_monoid M] (S : add_submonoid M) (N : Type*) [add_comm_monoid N]
 
 /-- The type of add_monoid homomorphisms satisfying the characteristic predicate: if `f : M →+ N`
     satisfies this predicate, then `N` is isomorphic to the localization of `M` at `S`. -/
-@[nolint has_inhabited_instance] structure localization_map :=
-(to_fun : M →+ N)
-(map_add_units : ∀ y : S, is_add_unit (to_fun y))
-(surj : ∀ z : N, ∃ x : M × S, z + to_fun x.2 = to_fun x.1)
-(eq_iff_exists : ∀ x y, to_fun x = to_fun y ↔ ∃ c : S, x + c = y + c)
+@[nolint has_inhabited_instance] structure localization_map
+extends add_monoid_hom M N :=
+(map_add_units' : ∀ y : S, is_add_unit (to_fun y))
+(surj' : ∀ z : N, ∃ x : M × S, z + to_fun x.2 = to_fun x.1)
+(eq_iff_exists' : ∀ x y, to_fun x = to_fun y ↔ ∃ c : S, x + c = y + c)
```

Comment:
Why you didn't need `map_add'` and `map_zero'` before?

Reply:
Sorry for being dense, but I don't understand the question.

Reply:
Ignore, I missed `+` in the old `to_fun`.